File: backend/common/models/task.go
package models

import (
	"time"
)

type Task struct {
	ID          int       `json:"id"`
	UserID      uint      `json:"user_id"`
	ProjectID   uint      `json:"project_id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	CreatedAt   time.Time `json:"created_at"`
	Status 		string 	  `json:"status"`
}

File: frontend/src/types/task.ts
export type TaskStatus = "to-do" | "in-progress" | "done";

export interface Task {
  id: number;
  title: string;
  description: string;
  status: TaskStatus;
  user_id: number;
  project_id: number;
  created_at: string;
}

export interface TaskResponse {
  status: string;
  message: string;
  data: {
    id: number;
    title: string;
    description: string;
    status: string; // still raw from backend
    user_id: number;
    project_id: number;
    created_at: string;
  };
}